extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/util/AuthZUtils.scala

```diff
@@ -84,6 +85,14 @@ private[authz] object AuthZUtils {
     getFieldVal[Option[CatalogTable]](plan, "catalogTable")
   }
 
+  def hasResolvedDatasourceV2Table(plan: LogicalPlan): Boolean = {
+    plan.nodeName == "DataSourceV2Relation" && plan.resolved
+  }
+
+  def getDatasourceV2Identifier(plan: LogicalPlan): Option[Identifier] = {
```

Comment:
If we use Identifier here, we need to update the doc that we do not support spark 2 anymore

Reply:
It is a possible downside for using it, that I didn't realized. Anything I could improve the compatibility?

Reply:
I don't whether it works or not, but something like this?
```scala
  def quoteIfNeeded(part: String): String = {
    if (part.matches("[a-zA-Z0-9_]+") && !part.matches("\\d+")) {
      part
    } else {
      s"`${part.replace("`", "``")}`"
    }
  }

  def quote(parts: Seq[String]): String = {
    parts.map(quoteIfNeeded).mkString(".")
  }

  def getDatasourceV2Identifier(plan: LogicalPlan): Option[TableIdentifier] = {
    val identifier = getFieldVal[Option[AnyRef]](plan, "identifier")
    identifier.map { id =>
      val namespaces = invoke(id, "namespaces").asInstanceOf[Array[String]]
      val table = invoke(id, "name").asInstanceOf[String]
      TableIdentifier(table, Some(quote(namespaces)))
    }
  }
```

Reply:
Good suggestion. Let me follow it.